app/src/main/java/by/academy/jc/ht/zhabko/AverageCounterThreads.java

```diff
@@ -0,0 +1,106 @@
+package by.academy.jc.ht.zhabko;
+
+
+import java.util.List;
+
+public class AverageCounterThreads {
+
+    List<Integer> sourceList;
+
+    private static class ThreadCounter extends Thread {
+        private int averageSum;
+        private int start;
+        private int end;
+        private  List<Integer> techSourceList;
+
+        private ThreadCounter (List <Integer> techSourceList, int start, int end) {
+            this.techSourceList = techSourceList;
+            this.start = start;
+            this.end = end;
+        }
+
+        private int getAverageSum() {
+            return averageSum;
+        }
+
+        @Override
+        public void run() {
+            for (int i = start; i < end; i++) {
```

Comment:
Verify `techSourceList` first of all. Throw `IllegalArgumentException` in a case of size mismatch or wrong indexes.

Cover exceptions with unit tests